src/ff/prime_field.rs

```diff
@@ -1,14 +1,15 @@
 use super::{field::BinaryField, Field};
-use serde::{Deserialize, Serialize};
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Not};
 
 macro_rules! field_impl {
-    ( $field:ty ) => {
+    ( $field:ty, $int:ty ) => {
         impl std::ops::Add for $field {
             type Output = Self;
 
             fn add(self, rhs: Self) -> Self::Output {
-                Self((self.0 + rhs.0) % Self::PRIME)
+                let c = u64::from;
+                debug_assert!(c(Self::PRIME) < (u64::MAX >> 1));
```

Comment:
I tried to use `const_assert!` from the `static_assertions` crate [docs](https://docs.rs/static_assertions/1.1.0/static_assertions/macro.const_assert.html) but it didn't work.

It basically objected to the use of `Self::` within that macro.